gradoop-common/src/test/java/org/gradoop/common/model/impl/properties/PropertyValueTest.java

```diff
@@ -15,389 +15,134 @@
  */
 package org.gradoop.common.model.impl.properties;
 
-import com.google.common.collect.Lists;
 import org.apache.flink.core.memory.DataInputView;
 import org.apache.flink.core.memory.DataOutputView;
-import org.gradoop.common.model.impl.id.GradoopId;
 import org.gradoop.common.exceptions.UnsupportedTypeException;
-import org.junit.Test;
+import org.gradoop.common.model.impl.id.GradoopId;
+import org.testng.annotations.Test;
 
 import java.io.IOException;
 import java.math.BigDecimal;
 import java.math.BigInteger;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.PriorityQueue;
-import java.util.Set;
+import java.util.*;
 
 import static org.gradoop.common.GradoopTestUtils.*;
 import static org.gradoop.common.model.impl.properties.PropertyValue.create;
-import static org.junit.Assert.*;
+import static org.testng.Assert.assertNotEquals;
+import static org.testng.AssertJUnit.*;
 
 public class PropertyValueTest {
 
   /**
-   * Tests if {@link PropertyValue#create(Object)} works with supported types.
+   * Tests if conversion to {@link BigDecimal} fails if incompatible type is given.
    */
+  @Test(expectedExceptions = UnsupportedOperationException.class,
+    dataProvider = "nonNumericalPropertyValueProvider",
+    dataProviderClass = PropertyValueTestProvider.class)
+  public void testUnsupportedBigDecimalConversionThrowsException(PropertyValue actual) {
```

Comment:
This is true, i dont see why this would invalidate the test though. The test checks whether `getBigDecimal` applied to any non numerical `PropertyValue` causes an `UnsupportedOperationException` to be thrown. I agree that the naming is a bit misleading since this exception should be raised in any case where `value.getType != X.class` and `value.getX` is issued.

Reply:
Then test it for all cases, not just for BigDecimal.